src/main/java/com/cloudbees/groovy/cps/tool/Translator.java

```diff
@@ -0,0 +1,814 @@
+import org.codehaus.groovy.runtime.DefaultGroovyMethods;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.ArrayType;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.NoType;
+import javax.lang.model.type.PrimitiveType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.TypeVariable;
+import javax.lang.model.type.WildcardType;
+import javax.lang.model.util.ElementScanner7;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.SimpleTypeVisitor6;
+import javax.lang.model.util.Types;
+import javax.tools.DiagnosticListener;
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import java.io.File;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Generates {@code CpsDefaultGroovyMethods} from the source code of {@code DefaultGroovyMethods}.
+ *
+ * @author Kohsuke Kawaguchi
+ */
+@SuppressWarnings("Since15")
+public class Translator {
+
+    private Types types;
+    private Elements elements;
+    private Trees trees;
+    private JavacTask javac;
+
+    private final JCodeModel codeModel = new JCodeModel();
+    private final JDefinedClass $CpsDefaultGroovyMethods;
+
+    // class references
+    private final JClass $Caller;
+    private final JClass $CpsFunction;
+    private final JClass $CpsCallableInvocation;
+    private final JClass $Builder;
+    private final JClass $CatchExpression;
+    private final JClass $DefaultGroovyMethods;
+
+    public Translator() {
+        try {
+            $CpsDefaultGroovyMethods = codeModel._class("com.cloudbees.groovy.cps.CpsDefaultGroovyMethods");
+        } catch (JClassAlreadyExistsException e) {
+            throw new AssertionError(e);
+        }
+        $Caller                = codeModel.ref("com.cloudbees.groovy.cps.impl.Caller");
+        $CpsFunction           = codeModel.ref("com.cloudbees.groovy.cps.impl.CpsFunction");
+        $CpsCallableInvocation = codeModel.ref("com.cloudbees.groovy.cps.impl.CpsCallableInvocation");
+        $Builder               = codeModel.ref("com.cloudbees.groovy.cps.Builder");
+        $CatchExpression       = codeModel.ref("com.cloudbees.groovy.cps.CatchExpression");
+        $DefaultGroovyMethods  = codeModel.ref("org.codehaus.groovy.runtime.DefaultGroovyMethods");
+    }
+
+    public static void main(String[] args) throws Exception {
+        new Translator().translate(new File("DefaultGroovyMethods.java"));
```

Comment:
I guess you did not read [my comment](https://github.com/cloudbees/groovy-cps/issues/7#issuecomment-203993887), or just did not care to address it now.